draft-ietf-taps-arch.md

```diff
@@ -216,35 +216,53 @@ Transport Services implementations are meant to be flexible at connection establ
+## Provide Common APIs for Common Features
 
-As a baseline, any Transport Services API needs to allow access to the distilled minimal set of features offered by transport protocols {{?I-D.ietf-taps-minset}}.
+Any functionality that is common across multiple transport protocols SHOULD be made accessible through a unified set of Transport Services API calls. As a baseline, any Transport Services API MUST allow access to the minimal set of features offered by transport protocols {{?I-D.ietf-taps-minset}}.
 
-## Access to Specialized Features
+An application can specify constraints and preferences for the protocols, features, and network interfaces it will use via Properties. A Transport Services API SHOULD offer Properties that are common to multiple transport protocols, in order to enable the system to appropriately select between protocols that offer equivalent features. Similarly, a Transport Services API SHOULD offer Properties that are applicable to a variety network layer interfaces and paths, in order to permit racing of different network paths without affecting the applications using the system. Each Property is expected to have a default value.
 
-There are applications that will need to control fine-grained details of transport protocols to optimize their behavior and ensure compatibility with remote systems. A Transport Services system therefore ought to also permit more specialized protocol features to be used. The interface for these specialized options ought to be exposed differently from the common options to ensure flexibility.
+Properties defaults SHOULD be selected to ensure correctness for the widest set of applications, while providing the widest set of options for selection. For example, since both applications that require data reliability and those that do not require data reliability can function correctly when a protocol provides reliability, reliability ought to be required by default. In contrast, the default value for a Property regarding the selection of network interfaces ought to permit as many interfaces as possible.
+
+Applications using a Transport Services system interface are REQUIRED to be robust to the automated selection provided by the system, given the ability to express requirements and preferences that constrain the choices that can be made by the system.
+
+## Allow Access to Specialized Features
+
+There are applications that will need to control fine-grained details of transport protocols to optimize their behavior and ensure compatibility with remote systems. A Transport Services system therefore SHOULD permit more specialized protocol features to be used. The interface for these specialized options ought to be exposed differently from the common options to ensure flexibility.
```

Comment:
I don't quite understand this yet:
What does "exposed differently" mean here? Is this about us not standardizing specific protocol selection and/or other specific features? Basically "make this harder to use so it doesn't become default and, thus, unnecessarily constrains selection"?

However: Couldn't one argue that some of the Selection Properties we have in the API draft are pretty specialized, too?
E.g., the User Timeout Option, which we give as an example below, is a Selection Property just like the others.

Reply:
Removed the sentence, as I agree it doesn't add much.

With regards to User Timeout Option, I think this argues that if an app prefers UTO, it should do a soft prefer or express "if you use TCP, you must use UTO" as opposed to simply requiring UTO (which would lead to just using TCP).